beacon-light-client/plonky2/circuits_executables/src/wrap_final_layer_in_poseidon_bn128.rs

```diff
@@ -0,0 +1,70 @@
+use std::fs;
+
+use plonky2::{
+    field::goldilocks_field::GoldilocksField,
+    fri::{reduction_strategies::FriReductionStrategy, FriConfig},
+    iop::witness::{PartialWitness, WitnessWrite},
+    plonk::{
+        circuit_builder::CircuitBuilder,
+        circuit_data::{CircuitConfig, CircuitData},
+        config::PoseidonGoldilocksConfig,
+        proof::ProofWithPublicInputs,
+    },
+};
+use redis::AsyncCommands;
+
+use crate::poseidon_bn128_config::PoseidonBN128GoldilocksConfig;
+
+pub async fn wrap_final_layer_in_poseidon_bn_128(
+    mut con: redis::aio::Connection,
+    compile_circuit: bool,
+    final_layer_circuit: CircuitData<GoldilocksField, PoseidonGoldilocksConfig, 2>,
+    final_layer_proof: ProofWithPublicInputs<GoldilocksField, PoseidonGoldilocksConfig, 2>,
+) -> Result<(), anyhow::Error> {
+    let standard_recursion_config = CircuitConfig::standard_recursion_config();
+    let config = CircuitConfig {
+        fri_config: FriConfig {
+            rate_bits: 6,
+            cap_height: 4,
+            proof_of_work_bits: 16,
+            reduction_strategy: FriReductionStrategy::ConstantArityBits(4, 5),
+            num_query_rounds: 14,
+        },
+        ..standard_recursion_config
+    };
+    let mut builder = CircuitBuilder::<GoldilocksField, 2>::new(config);
+    let verifier_target = builder.constant_verifier_data(&final_layer_circuit.verifier_only);
+    let proof_target = builder.add_virtual_proof_with_pis(&final_layer_circuit.common);
+    builder.verify_proof::<PoseidonGoldilocksConfig>(
+        &proof_target.clone(),
+        &verifier_target,
+        &final_layer_circuit.common,
+    );
+    builder.register_public_inputs(&proof_target.public_inputs);
+    let mut pw = PartialWitness::new();
+    pw.set_proof_with_pis_target(&proof_target, &final_layer_proof);
+    let circuit_data = builder.build::<PoseidonBN128GoldilocksConfig>();
+    let proof = circuit_data.prove(pw)?;
+    let proof = serde_json::to_string(&proof)?;
```

Comment:
Try making this a single line or rename the original proof

Reply:
Let's switch to using the `CommandLineOptionsBuilder` for `final_layer` in this PR

Reply:
I would use the name `serialized_proof` or `marshalled_proof`